src/deploy/functions/build.ts

```diff
@@ -251,32 +275,30 @@ export function resolveBackend(build: Build): backend.Backend {
 
       const bkEndpoint: backend.Endpoint = {
         id: endpointId,
-        project: "",
+        project: endpoint.project,
         region: region,
         entryPoint: endpoint.entryPoint,
         platform: endpoint.platform,
-        runtime: "",
-        labels: endpoint.labels,
-        environmentVariables: endpoint.environmentVariables,
-        secretEnvironmentVariables: undefined,
-        availableMemoryMb: endpoint.availableMemoryMb,
+        runtime: endpoint.runtime,
         timeoutSeconds: timeout,
         ...trigger,
       };
       proto.renameIfPresent(bkEndpoint, endpoint, "maxInstances", "maxInstances", resolveInt);
       proto.renameIfPresent(bkEndpoint, endpoint, "minInstances", "minInstances", resolveInt);
       proto.renameIfPresent(bkEndpoint, endpoint, "concurrency", "concurrency", resolveInt);
       proto.copyIfPresent(bkEndpoint, endpoint, "ingressSettings");
+      proto.copyIfPresent(bkEndpoint, endpoint, "availableMemoryMb");
```

Comment:
FYI, if you really truly can copy, `copyIfPresent` is variadic and you can say

```
proto.copyIfPresent(
  bkEndpiont,
  endpiont,
  "ingressSettings",
  "availableMemoryMb",
  "environmentVariables",
  "labels",
);
```

Reply:
Done.